base/compiler/ssair/inlining.jl

```diff
@@ -831,29 +831,44 @@ function compileable_specialization(mi::MethodInstance, effects::Effects,
     return InvokeCase(mi_invoke, effects, info)
 end
 
+function compileable_specialization(ci::CodeInstance, effects::Effects,
+        et::InliningEdgeTracker, @nospecialize(info::CallInfo); compilesig_invokes::Bool=true)
+    if compilesig_invokes ? !isa_compileable_sig(ci.def.specTypes, ci.def.sparam_vals, ci.def.def) :
+            any(@nospecialize(t)->isa(t, TypeVar), ci.def.sparam_vals)
+        return compileable_specialization(ci.def, effects, et, info; compilesig_invokes)
+    end
+    add_inlining_backedge!(et, ci.def) # to the dispatch lookup
+    push!(et.edges, ci.def.def.sig, ci.def) # add_inlining_backedge to the invoke call
```

Comment:
I think this may need fixing now, to match the fix to the other case

Reply:
Since this code path never changes the mi, should this call just be deleted then?

Reply:
Unclear, since the inlining code seems to be required to add an edge here to `ci` itself, but such an edge is not currently valid to create, mainly due to the unresolved semantic questions mentioned in https://hackmd.io/@Og2_pcUySm6R_RPqbZ06JA/S1bqP1D_6

So yeah, maybe just drop it, since this PR is just adding the functionality, but not defining the behavior yet?